js/models/article.js

```diff
@@ -0,0 +1,35 @@
+'use strict';
+
+(function (exports) {
+  exports.Article = function (headline, url, image, summary) {
+    // headline = headline;
+    // url = url;
+    // image = image;
+    // summary = summary;
+
+    function showHeadline() {
+      return headline;
+    }
+
+    function showSummary() {
+      return summary;
+    }
+
+    function showUrlToFullStory() {
+      return url;
+    }
+
+    function featureImage() {
+      return image;
+    }
+
+    return {
+      showHeadline: showHeadline,
+      showSummary: showSummary,
+      showUrlToFullStory: showUrlToFullStory,
+      featureImage: featureImage
+    };
+
+    var headline, url, image, summary
```

Comment:
I'd be interested to see the article you were reading on variable hoisting. I think there's a difference between variable hoisting as something which javascript does, and the pattern in which you have written this code to take advantage of that. In the case of the latter, I feel a bit uneasy about it and it's not clear to me what benefits it has.